translator/trace/zipkin/status_code.go

```diff
@@ -55,6 +57,21 @@ func (m *statusMapper) ocStatus() *tracepb.Status {
 		s = m.fromHTTP
 	}
 
+	// If no codePtr was provided, fallback to the first source with a message
+	// and use the error tag code
+	if s.codePtr == nil {
+		switch {
+		case m.fromCensus.message != "":
+			s = m.fromCensus
+		case m.fromStatus.message != "":
+			s = m.fromStatus
+		default:
+			s = m.fromHTTP
+		}
+
+		s.codePtr = m.fromErrorTag.codePtr
```

Comment:
I think this should be under condition `if (s.codePtr == nil)` so we don't override code that was already set

Reply:
I'll add it to make it more clear, but this would always be true currently

Reply:
I don't understand why this always be true? any of the assignments above can change it not null, no?

Reply:
From the first switch we know that
m.fromCensus.codePtr == nil
m.fromStatus.codePtr == nil

and line 62 gives
m.fromHTTP.codePtr == nil

So we've verified all three are nil

Reply:
Sorry, I still don't get it. `s` can be set in any `case` above, so any of them can set `s.codePtr` to be something but not `nil`, right? If we don't have `if (s.codePtr == nil)` condition here, we override any `s.codePtr` with `m.fromErrorTag.codePtr` all the time